module-2/backend/week_two.md

```diff
@@ -5,37 +5,93 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+  - status code, header, body
 13. `Rack::Test` allows us to test our controllers in isolation. What are some of the methods it gives us to simulate the request/response cycle?
+  - you can use the regular http verbs (get/put/post/delete) to make requests and then use things like .last_response to check and see if the right data is coming back
 14. Describe some techniques to make our Sinatra views more DRY. Give an example of when you would use these techniques.
+  - building logic out into partials, for example if you're repeating a form many times over the course of an application.
+  - using a layout.erb file to build the basic structure for your views - if you've got a header or a navbar or even just an app-wide CSS file you could use a layout file to make sure each page loads the same styling
 
 
 ### Optional Questions
 
 1. Name your five favorite ActiveRecord methods (i.e. methods your models inherit from ActiveRecord) and describe what they do.
+  - .create - makes a thing, saves that thing
+  - .pluck - finds a thing in a table by column names
+  - .average - returns average value of column contents
+  - .order - lets you specify the way returned results should be ordered
+  - .limit - allows you to specify exactly how many records should be returned
 2. Name your three favorite ActiveRecord rake tasks and describe what they do.
+  - rake db:drop - blows up the database. scorched-earth. nothing's left. time to rebuild.
+  - rake db:create - with a word we create... a database! (or more!)
+  - rake db:migrate - sets everything up, makes changes, etc.
 3. What's the difference between agile workflow and waterfall method?
+  - see q# 6 above.
 4. What can you expect from a group as you begin working together? As you continue working together?
+  - awkwardness as you begin, maybe some "storming" or conflict. Eventually if all goes well: teamwork, success.
 5. What two columns does `t.timestamps null: false` create in our database?
+  - it creates created_at and updated_at columns
 6. What cURL flag can you use to send a `POST` request?
+  - curl -d
 7. What case does JSON (and JavaScript) use for multi-word variables?
+  - camel-case
 8. What case does Ruby use for multi-word variables?
+  - snake-case
 9. In a database that's holding schools and teachers, what will be the relationship between schools and teachers?
+  - schools will have_many teachers, teachers will belong_to schools.
 10. In the same database, what will you need to do to create this relationship (draw a schema diagram)?
-11. Give an example of when you might want to store information besides ids on a join table.
+    School - > has_many Teachers
+    (id, name)
+    Teacher - > belongs_to School
+    (id, name, school_id)
+11. Give an example of when you might want to store information besides ids on a join table.  
+  you might want to store info besides ids on a join table if objects have other unique identifiers - SSN's, VINs, etc
```

Comment:
Maybe? I think the key to making this decision would be if there's some data that's specific to the join that you would want to save. For example: assume you have a store where prices change all the time, but you want to make sure that you have a historical record of how much someone paid for a specific transaction. Assuming a transaction is represented as a join between `invoice` and `item` (with a user having many invoices), I would want to store the price for each item on the `invoice_items` table to freeze that price to the price at the time the invoice was created.